lib/ansible/modules/windows/win_share.ps1

```diff
@@ -83,7 +83,13 @@ If ($state -eq "absent") {
     }
 
     # normalize path and remove slash at the end
-    $path = (Get-Item $path).FullName -replace "\\$"
+    $path = (Get-Item $path).FullName
+
+    # Allow for X:\ case
+    If ("$path" -Match "(.+)[^\:]\\$")
```

Comment:
This will strip off the character before a trailing slash so `C:\Windows\` becomes `C:\Window` which we obviously don't want. To do what you want you want to move `[^\:]` so that it still doesn't match `:` if it becomes before `\` but still include it in the final match group. The only other thing I think we should support is if someone specified `path: C:` in their task we should automatically change it to `C:\` for them.

Reply:
Fixed and i will add something for the C: case